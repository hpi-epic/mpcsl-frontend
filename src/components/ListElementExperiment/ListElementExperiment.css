@import '~antd/dist/antd.css';

.Card {
	width: 50%;
	margin: 0 auto;
	border-radius: 8px;
}

.List-Buttons {
	float: right;
	margin-right: 5px;
}

.Card-Title {
	float: left;
	background-color: transparent;
	border: none;
	margin-top: 0;
	margin-bottom: .5em;
	color: rgba(0, 0, 0, 0.85);
	font-weight: 500;
	display: block;
	font-size: 1em;
	margin-block-start: 0.83em;
	margin-block-end: 0.83em;
	margin-inline-start: 0px;
	margin-inline-end: 0px;
	outline: none;
}

.Card-Badge {
	float: right;
	margin-right: 5px;
	margin-top: 15px;
}

.Card-Content {
	float: left;
}

.Dropdown-Button {
	border: none;
	background-color: transparent;
}

.Dropdown-Button:hover {
	border: none;
	background-color: transparent;
}

.Go-Back-Button {
	float: left;
	margin-right: 20px;
}

