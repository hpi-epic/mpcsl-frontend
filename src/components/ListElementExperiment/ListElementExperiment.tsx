import { Badge, Button, Card, Dropdown, List, Menu } from 'antd';
import { BadgeProps } from 'antd/lib/badge';
import React from 'react';
import './ListElementExperiment.css';

interface IPropsListElementExperiment {
  title: string;
  status?: BadgeProps['status'];
  statusText?: string;
  content: string;
  onDuplicate: (e: React.MouseEvent<HTMLElement>) => void;
  onDelete: (e: React.MouseEvent<HTMLElement>) => void;
  onRunStart: (e: React.MouseEvent<HTMLElement>) => void;
  onExplore: (e: React.MouseEvent<HTMLElement>) => void;
  onView: (e: React.MouseEvent<HTMLElement>) => void;
  showAllJobs: (e: React.MouseEvent<HTMLElement>) => void;
}

function ListElementExperiment(props: IPropsListElementExperiment) {
  const menu = (
    <Menu>
      <Menu.Item>
        <Button className='Dropdown-Button' onClick={props.onDelete}>
          Delete
        </Button>
      </Menu.Item>
      <Menu.Item>
        <Button className='Dropdown-Button' onClick={props.onDuplicate}>
          Duplicate
        </Button>
      </Menu.Item>
      <Menu.Item>
        <Button className='Dropdown-Button' onClick={props.showAllJobs}>
          Jobs
        </Button>
      </Menu.Item>
    </Menu>
  );

  const cardTitle = (
    <div>
      <h2 className='Card-Title'>{props.title}</h2>
      <Badge
        className='Card-Badge'
        status={props.status}
        text={props.statusText}
      />
    </div>
  );

  return (
    <div>
      <List.Item>
        <Card className='Card' title={cardTitle}>
          <p className='Card-Content'>{props.content}</p>
          <Dropdown overlay={menu} placement='bottomLeft'>
            <Button className='List-Buttons' icon='ellipsis' />
          </Dropdown>
          <Button className='List-Buttons' onClick={props.onView}>
            View
          </Button>
          <Button className='List-Buttons' onClick={props.onRunStart}>
            Run
          </Button>
          <Button className='List-Buttons' onClick={props.onExplore}>
            Explore
          </Button>
        </Card>
      </List.Item>
    </div>
  );
}

export default ListElementExperiment;
