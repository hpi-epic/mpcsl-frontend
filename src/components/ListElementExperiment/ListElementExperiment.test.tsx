import React from 'react';
import renderer from 'react-test-renderer';
import ListElementExperiment from './ListElementExperiment';

describe('<ListElementExperiment />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ListElementExperiment
          title='Test Title'
          status='success'
          statusText='done'
          content='Test Content'
          onDelete={() => undefined}
          onDuplicate={() => undefined}
          onRunStart={() => undefined}
          onExplore={() => undefined}
          onView={() => undefined}
          onShowDetails={() => undefined}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
