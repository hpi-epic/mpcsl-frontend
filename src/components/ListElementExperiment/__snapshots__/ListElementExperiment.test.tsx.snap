// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListElementExperiment /> renders correctly 1`] = `
<div>
  <div
    className="ant-list-item"
  >
    <div
      className="ant-list-item-content ant-list-item-content-single"
    >
      <div
        className="ant-card Card ant-card-bordered"
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              <div>
                <button
                  className="Card-Title"
                  onClick={[Function]}
                >
                  Test Title
                </button>
                <span
                  className="Card-Badge ant-badge ant-badge-status ant-badge-not-a-wrapper"
                >
                  <span
                    className="ant-badge-status-dot ant-badge-status-success"
                  />
                  <span
                    className="ant-badge-status-text"
                  >
                    done
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <p
            className="Card-Content Experiment-Content"
          >
            Test Content
          </p>
          <div>
            <button
              className="ant-btn List-Buttons ant-dropdown-trigger ant-btn-icon-only"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              type="button"
            >
              <i
                className="anticon anticon-ellipsis"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="ellipsis"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0z"
                  />
                </svg>
              </i>
            </button>
            <button
              className="ant-btn List-Buttons ant-btn-primary ant-btn-background-ghost"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                Explore
              </span>
            </button>
            <button
              className="ant-btn List-Buttons ant-btn-primary ant-btn-background-ghost"
              onClick={[Function]}
              type="button"
            >
              <span>
                Run
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
